Compilation errors [gcc 10.1, Linux 5.6]

When I run

```
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX="$HOME/.local/" ..
make -j4
```

I get this error:

```
In file included from /home/rk/z/flatbuffers/tests/test.cpp:39:
/home/rk/z/flatbuffers/tests/arrays_test_generated.h: In constructor ‘MyGame::Example::ArrayStruct::ArrayStruct(float, int8_t, int32_t)’:
/home/rk/z/flatbuffers/tests/arrays_test_generated.h:156:34: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘struct MyGame::Example::NestedStruct’; use assignment or value-initialization instead [-Werror=class-memaccess]
  156 |     std::memset(d_, 0, sizeof(d_));
      |                                  ^
/home/rk/z/flatbuffers/tests/arrays_test_generated.h:66:40: note: ‘struct MyGame::Example::NestedStruct’ declared here
   66 | FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(8) NestedStruct FLATBUFFERS_FINAL_CLASS {
      |                                        ^~~~~~~~~~~~
```

and this one:

```
In file included from /home/rk/z/flatbuffers/include/flatbuffers/flatbuffers.h:20,
                 from /home/rk/z/flatbuffers/tests/test.cpp:18:
In function ‘void flatbuffers::WriteScalar(void*, T) [with T = unsigned char]’,
    inlined from ‘void flatbuffers::Vector<T>::Mutate(flatbuffers::uoffset_t, const T&) [with T = unsigned char]’ at /home/rk/z/flatbuffers/include/flatbuffers/flatbuffers.h:314:16,
    inlined from ‘void MutateFlatBuffersTest(uint8_t*, std::size_t)’ at /home/rk/z/flatbuffers/tests/test.cpp:465:20:
/home/rk/z/flatbuffers/include/flatbuffers/base.h:381:29: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
  381 |   *reinterpret_cast<T *>(p) = EndianScalar(t);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```

This happens in the current master (6df40a24) and  v1.12.0
[make.log](https://github.com/google/flatbuffers/files/4739704/make.log)
[cmake.log](https://github.com/google/flatbuffers/files/4739705/cmake.log)

--- a/include/flatbuffers/base.h
+++ b/include/flatbuffers/base.h
@@ -381,6 +381,13 @@ T ReadScalar(const void *p) {
   return EndianScalar(*reinterpret_cast<const T *>(p));
 }
 
+// See https://github.com/google/flatbuffers/issues/5950
+
+#if (FLATBUFFERS_GCC >= 100000) && (FLATBUFFERS_GCC < 110000)
+  #pragma GCC diagnostic push
+  #pragma GCC diagnostic ignored "-Wstringop-overflow"
+#endif
+
 template<typename T>
 // UBSAN: C++ aliasing type rules, see std::bit_cast<> for details.
 __supress_ubsan__("alignment")
@@ -392,6 +399,10 @@ template<typename T> struct Offset;
 template<typename T> __supress_ubsan__("alignment") void WriteScalar(void *p, Offset<T> t) {
   *reinterpret_cast<uoffset_t *>(p) = EndianScalar(t.o);
 }
+
+#if (FLATBUFFERS_GCC >= 100000) && (FLATBUFFERS_GCC < 110000)
+  #pragma GCC diagnostic pop
+#endif
 
 // Computes how many bytes you'd have to pad to be able to write an
 // "scalar_size" scalar if the buffer had grown to "buf_size" (downwards in

